### PR TITLE
Implement REST API for Resident IDP Inbound Authentication Configurations

### PR DESCRIPTION
**Pull request by ThaminduDilshan**
## Purpose
Implement a REST API to retrieve/ update the inbound authentication configurations of the resident IDP. REST API is provided for the following configs.

Screenshot from 2023-10-10 18-55-48

## Related issue
- https://github.com/wso2/product-is/issues/16911

## Related PR
- https://github.com/wso2/carbon-identity-framework/pull/4987

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -730,7 +730,7 @@
         <maven.buildnumber.plugin.version>1.4</maven.buildnumber.plugin.version>
         <org.apache.felix.annotations.version>1.2.4</org.apache.felix.annotations.version>
         <identity.governance.version>1.8.62</identity.governance.version>
-        <carbon.identity.framework.version>5.25.337</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.25.375-SNAPSHOT</carbon.identity.framework.version>
```
Comment (ThaminduDilshan) [inline pom.xml:733]: TODO: Bump framework version

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java [hunk 2]
```diff
@@ -110,6 +114,12 @@ public class ServerConfigManagementService {
 
     private static final Log log = LogFactory.getLog(ServerConfigManagementService.class);
 
+    private static final List<String> FEDERATED_AUTHENTICATORS_TO_UPDATE = new ArrayList<>(Arrays.asList(
+            IdentityApplicationConstants.Authenticator.SAML2SSO.NAME,
+            IdentityApplicationConstants.Authenticator.PassiveSTS.NAME,
+            IdentityApplicationConstants.Authenticator.OIDC.NAME)
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:120]: We haven't changed the OIDC configs, right?

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java [hunk 3]
```diff
@@ -110,6 +114,12 @@ public class ServerConfigManagementService {
 
     private static final Log log = LogFactory.getLog(ServerConfigManagementService.class);
 
+    private static final List<String> FEDERATED_AUTHENTICATORS_TO_UPDATE = new ArrayList<>(Arrays.asList(
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:117]: Though we use federated authenticators to represent the configs in the resident IDP, naming the constant like this adds confusions IMO.

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java [hunk 4]
```diff
@@ -1181,4 +1191,287 @@ private APIError handleNotFoundError() {
...
+                if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                    samlMetadataEndpoint = ServiceURLBuilder.create().addPath(
+                            Constants.SAML2_METADATA_ENDPOINT_URI_PATH).build().getAbsolutePublicURL();
+                } else {
+                    samlMetadataEndpoint = ServiceURLBuilder.create().addPath("/t/" + tenantDomain +
+                            Constants.SAML2_METADATA_ENDPOINT_URI_PATH).build().getAbsolutePublicURL();
+                }
+                inboundAuthConfig.setMetadataEndpoint(samlMetadataEndpoint);
+            } else {
+                throw handleException(Response.Status.NOT_FOUND,
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:1247]: Not being able to find the resident IDP should be a server error IMO

Comment (madurangasiriwardena) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:1247]: Should be same for similar errors in other places.

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java (hunk 2 above)
Comment (ThaminduDilshan) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:120]: Removed. Logic is updated to only change the required federated authenticator

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java (hunk 3 above)
Comment (ThaminduDilshan) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java]: Removed. Logic is updated to only change the required federated authenticator

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java (hunk 4 above)
Comment (ThaminduDilshan) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java]: Fixed

File: pom.xml (hunk 1 above)
Comment (ThaminduDilshan) [reply pom.xml]: Done

**Review by madurangasiriwardena — APPROVED** [review]

**Pull request merged**